### PR TITLE
CNS-792: Timerstore module README

### PR DESCRIPTION
**Pull request by shleikes**

### CODE REVIEW

File: x/timerstore/README.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+The `WithCallbackByBlockHeight` function will add the callback to the `TimerStore` object, under the `BlockHeight` kind, and the `WithCallbackByBlockTime` function will add the callback to the `TimerStore` object, under the `BlockTime` kind.
+
+The `BlockHeight` callbacks, can be triggered at a given block, and the `BlockTime` callback can be triggered at the first block that it's time is past the given time.
+
+### Trigger
+
+As stated in the previous section, timers can be one of two kinds: `BlockHeight` or `BlockTime`.  
+Once a callback is created in the TimerStore, a procedure can then create a new timer corresponding to its kind.
+
+For `BlockHeight` callbacks, the function `AddTimerByBlockHeight(ctx sdk.Context, block uint64, key, data []byte)` will create a timer that will trigger at `block`, with the given `key` and `data`.
```
Comment (oren-lava) [inline x/timerstore/README.md:61]: emphasize the the `key` argument is actually the timer's subkey, so it doesn't have to be unique (the timers' primary key is unique and is created inside the timerstore module). Also, note that the `data` argument is useful for attaching data to the timer that will be available when the callback function is triggered

File: x/timerstore/README.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+## Queries
+
+The `timerstore` module supports the following queries:
+
+| Query        | Arguments                           | What it does                                     |
+| ------------ | ----------------------------------- | ------------------------------------------------ |
+| `all-timers` | store-key (string), prefix (string) | Shows all timers of a specific timer store       |
+| `next`       | store-key (string), prefix (string) | Shows the next timeout of a specific timer store |
+| `store-keys` | none                                | Shows all timer store keys                       |
```
Comment (oren-lava) [inline x/timerstore/README.md:79]: all timer store keys **and prefixes**

File: x/timerstore/client/cli/query_store_keys.go [hunk 3]
```diff
@@ -14,7 +14,7 @@ var _ = strconv.Itoa(0)
 func CmdStoreKeys() *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "store-keys",
-		Short: "Query for all timer store keys",
+		Short: "Query all timer store keys",
```
Comment (oren-lava) [inline x/timerstore/client/cli/query_store_keys.go:17]: and prefixes

**Review by oren-lava — CHANGES_REQUESTED** [review]

File: x/timerstore/README.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
+# `x/timerstore`
+
+## Abstract
+
+This document specifies the `timerstore` module of Lava Protocol.
+
+This module, primarily a utility for other modules, is not a standard module with which users can interact, except for a few queries intended for debugging purposes. The module is used as a powerful utility for other modules.
+
+The timerstore allows other modules to create timers, that triggers a callback functions.
```
[the remark this answers is not in the document]
Comment (shleikes) [reply x/timerstore/README.md:9]: Fixed in 3e28660ee7800b52389e525aef295f0f973ceb5d

File: x/timerstore/README.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
+# `x/timerstore`
+
+## Abstract
+
+This document specifies the `timerstore` module of Lava Protocol.
+
+This module, primarily a utility for other modules, is not a standard module with which users can interact, except for a few queries intended for debugging purposes. The module is used as a powerful utility for other modules.
```
[the remark this answers is not in the document]
Comment (shleikes) [reply x/timerstore/README.md]: Fixed in 0908c66206673f84f2c211afe9e75ce361fbc743

File: x/timerstore/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (shleikes) [reply x/timerstore/README.md]: Fixed in 3e28660ee7800b52389e525aef295f0f973ceb5d

File: x/timerstore/README.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+The timerstore allows other modules to create timers, that triggers a callback functions.  
+A timer defined in this module can be of two types: BlockHeight or BlockTime.
```
[the remark this answers is not in the document]
Comment (shleikes) [reply x/timerstore/README.md]: Fixed in 29fdcf8b08e0915b8466a693da7bce139b0bf755

File: x/timerstore/README.md [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+### BeginBlock & EndBlock
+
+A module can decide whether to trigger a certain timer at the start of the block or at the end of it.  
+The function `NewTimerStoreBeginBlock` will create a new `TimerStore` that will trigger at the `BeginBlock` when the time is right, and the function `NewTimerStoreEndBlock` will create a new `TimerStore` that will trigger at the `EndBlock` when the time is right.
+
+### BlockHeight & BlockTime
+
+After calling the function `NewTimerStoreBeginBlock` or `NewTimerStoreEndBlock` the calling procedure can use the returned `TimerStore` object to set the callback of the timer. The callback has to be of the signature `func(ctx sdk.Context, key, data []byte)`, more on that in the next section.
+
+To set the callback, the procedure needs to call `WithCallbackByBlockHeight` or `WithCallbackByBlockTime`.
```
[the remark this answers is not in the document]
Comment (shleikes) [reply x/timerstore/README.md]: Added snipper in 376efeb0cf937862f73d4c48045dda91eaec5f65

File: x/timerstore/README.md [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+The `WithCallbackByBlockHeight` function will add the callback to the `TimerStore` object, under the `BlockHeight` kind, and the `WithCallbackByBlockTime` function will add the callback to the `TimerStore` object, under the `BlockTime` kind.
+
+The `BlockHeight` callbacks, can be triggered at a given block, and the `BlockTime` callback can be triggered at the first block that it's time is past the given time.
+
+### Trigger
+
+As stated in the previous section, timers can be one of two kinds: `BlockHeight` or `BlockTime`.
```
[the remark this answers is not in the document]
Comment (shleikes) [reply x/timerstore/README.md]: Hope this is better 3f29fca4d0ffa1ccc68834152285cee30a8135d9

File: x/timerstore/README.md (hunk 1 above)
Comment (shleikes) [reply x/timerstore/README.md]: Clarified in 620343f3452b5248733823547ecdc6d7fc62010c

File: x/timerstore/README.md (hunk 2 above)
Comment (shleikes) [reply x/timerstore/README.md]: Fix in 1c3e89403e64df0743f3d8ea1cf3d33c0bfd5627

File: x/timerstore/client/cli/query_store_keys.go (hunk 3 above)
Comment (shleikes) [reply x/timerstore/client/cli/query_store_keys.go]: Fixed in 1c3e89403e64df0743f3d8ea1cf3d33c0bfd5627

File: x/timerstore/README.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (oren-lava) [reply x/timerstore/README.md]: much much better!

**Review by oren-lava — APPROVED** [review]

**Pull request merged**